test/flows/plan_adoption_leave_flow_test.rb

```diff
@@ -0,0 +1,77 @@
+require "test_helper"
+require "support/flow_test_helper"
+
+class PlanAdoptionLeaveFlowTest < ActiveSupport::TestCase
+  include FlowTestHelper
+
+  setup { testing_flow PlanAdoptionLeaveFlow }
+
+  should "render a start page" do
+    assert_rendered_start_page
+  end
+
+  context "question: child_match_date?" do
+    setup { testing_node :child_match_date? }
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node of child_arrival_date? for a response of 3 months ago" do
+        assert_next_node :child_arrival_date?, for_response: 3.months.ago.to_s
+      end
+    end
+  end
+
+  context "question: child_arrival_date?" do
+    setup do
+      testing_node :child_arrival_date?
+      add_responses child_match_date?: 3.months.ago.to_s
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node of leave_start? for a response that is after the child_match_date" do
+        assert_next_node :leave_start?, for_response: 1.day.after(3.months.ago).to_s
+      end
+
+      context "validation" do
+        should "be invalid if the response is the same as the child_match_date" do
+          assert_invalid_response 3.months.ago.to_s
+        end
+
+        should "be invalid if the response is before the child_match_date" do
+          assert_invalid_response 1.day.before(3.months.ago).to_s
+        end
+      end
+    end
+  end
+
+  context "question: leave_start?" do
+    setup do
+      testing_node :leave_start?
+      add_responses child_match_date?: 3.months.ago.to_s,
+                    child_arrival_date?: 3.months.since.to_s
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node of adoption_leave_details for a response that is within 14 days of the child_arrival_date" do
+        assert_next_node :adoption_leave_details, for_response: 14.days.ago(3.months.since).to_s
```

Comment:
The `14.days.ago(3.months.since)` is quite tough to understand. I think it'd better with a separate variable a la:

```suggestion
        adoption_leave_start = 3.months.since - 14.days
        assert_next_node :adoption_leave_details, for_response: adoption_leave_start.to_s
```